keras_cv/visualization/segmentation_mask.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility function to help visualize batch of binary mask.
+"""
+from collections import defaultdict
+
+import tensorflow as tf
+
+from keras_cv.visualization.colors import colors
+
+RGB = 3
+
+
+def _blend(image1, image2, factor=0.4):
+    difference = image2 - image1
+    scaled = factor * difference
+    # Do addition in float.
+    blended = image1 + scaled
+
+    # Interpolate
+    if factor > 0.0 and factor < 1.0:
+        # Interpolation means we always stay within 0 and 255.
+        return tf.round(blended)
+    # We need to clip and then cast.
+    blended = tf.round(tf.clip_by_value(blended, 0.0, 255.0))
+    return blended
+
+
+def _map_color_on_mask(masks, color_map):
+
+    # check distinct mask color codes with color mapping.
+    distinct_mask_code = tf.unique(tf.reshape(masks, -1)).y
+    if any([code not in distinct_mask_code for code in color_map.keys()]):
+        raise ValueError(
+            f"Color mapping {color_map} does not map completely with distint color codes present in masks: {distinct_mask_code}"
+        )
+
+    color_rgb = defaultdict(tuple)
+
+    # map color code with RGB
+    for code, color in color_map.items():
```

Comment:
@LukeWood 
replace this with 
tf.nest.map_structured(lambda color: colors.get(color), color_map)

but if the passed color is not found, its difficult to print a proper exception with this.